### PR TITLE
Create RadioButton, make composable, storytime

### PR DESCRIPTION
**Pull request by mosch**
I've made changes to the RadioButtonSet like we discussed. [Check it out in the storybook.](https://deploy-preview-302--allthings-elements.netlify.com/?selectedKind=Forms&selectedStory=RadioButton&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel)
I still need to think of a good way to test this. I need these components soon, so I did some hands on.

Checklist:

- [ ] Test added / Snapshots updated
- [ ] Documentation added / updated

### CODE REVIEW

File: src/molecules/RadioButton.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    } = this.props
+
+    const realId = id || value
+
+    return (
+      <Theme>
+        {({ theme, colorize }) => (
+          <View {...styles.radioElement} {...props}>
+            <Relative direction="row" alignV="center" alignH="center">
+              <View htmlElement="label" htmlFor={realId}>
```
Comment (dan-lee) [inline src/molecules/RadioButton.jsx]: If you wrap an input with a label you don't need an ID, so you could just remove it.

File: src/molecules/RadioButtonSet.test.js [hunk 2]
```diff
@@ -54,4 +61,4 @@ test('First radio-button should be selected', () => {
       .prop('defaultChecked')
   ).toBeTruthy()
   wrapper.unmount()
-})
+})*/
```
Comment (adashst) [inline src/molecules/RadioButtonSet.test.js]: Maybe this can be removed?

File: src/molecules/RadioButtonSet.jsx [hunk 3]
```diff
@@ -72,127 +43,88 @@ class RadioButtonSet extends React.Component {
...
-  }
+  static defaultProps = {}
 
   state = {
-    visible: true,
-    selected: this.props.defaultValue,
+    value: this.props.defaultValue,
   }
 
+  radios = []
```
Comment (adashst) [inline src/molecules/RadioButtonSet.jsx]: Afaiks this is an array of cloned children but what for?

**Comment by adashst** [comment]
There was also the idea to enable customised direction of the RadioButtons in the RadioButtonSet.

```
...
const DIRECTION_HORIZONTAL = 'horizontal'
const DIRECTION_VERTICAL = 'vertical'
const DIRECTION_AUTO = 'auto'  

class RadioButtonSet extends React.Component {  

    ...

    static propTypes = {
        ...
        /** Customize direction */
        direction: PropTypes.oneOf([
          DIRECTION_HORIZONTAL,
          DIRECTION_VERTICAL,
          DIRECTION_AUTO,
        ]),
     }  

     static defaultProps = {
         ...
         direction: DIRECTION_AUTO,
     }   

     getAutoDirection = () => (this.props.children.length > 2 ? 'column' : 'row')
     getDirection = () =>
         this.props.direction === DIRECTION_HORIZONTAL ? 'row' : 'column'

     ...

     render() {
        ...
        const direction =
           this.props.direction === DIRECTION_AUTO
           ? this.getAutoDirection()
           : this.getDirection()  

        return (
           <Theme>
              <View
                  direction={direction}>
                  <RadioButton ...>
                  ...
              </View>
           </Theme>
       )
    }
}
```

**Comment by adashst** [comment]
@mosch @dan-lee Added customisable direction: https://deploy-preview-302--allthings-elements.netlify.com/?selectedKind=Forms&selectedStory=RadioButton&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

File: src/molecules/RadioButtonSet.test.js (hunk 2 above)
Comment (mosch) [reply src/molecules/RadioButtonSet.test.js]: my test are not working, so we should maybe keep the existing ones?

**Pull request merged**